File: simulations.py
import random as rand
import time as t
import sys

def simulate(killcount, droprate):
    amount = 0
    for _ in range(killcount):
        if rand.randint(1, droprate) == 1:
            amount += 1
    return amount

def main():
    amount = int(sys.argv[1])
    for i in range(amount):
        start_time = t.time_ns()
        simulate(1_000_000, 256)
        end_time = t.time_ns()
        print(end_time - start_time)

if __name__ == '__main__':
    main()

File: simulations/simulations.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int simulate(int killcount, int droprate) {
    int amount = 0;
    for (int i = 0; i < killcount; i++) {
        if (rand() % droprate == 0) {
            amount++;
        }
    }
    return amount;
}

int main(int argc, char *argv[]) {
    int amount = atoi(argv[1]);
    srand(time(NULL));

    for (int i = 0; i < amount; i++) {
        struct timespec start_time, end_time;
        clock_gettime(CLOCK_MONOTONIC_RAW, &start_time);
        simulate(1000000, 256);
        clock_gettime(CLOCK_MONOTONIC_RAW, &end_time);
        long long elapsed_ns = (end_time.tv_sec - start_time.tv_sec) * 1000000000LL + (end_time.tv_nsec - start_time.tv_nsec);
        printf("%lld\n", elapsed_ns);
    }

    return 0;
}
